dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/trackedentity/AbstractTrackedEntityInstanceService.java

```diff
@@ -1198,18 +1188,28 @@ private TrackedEntityInstance getTei( org.hisp.dhis.trackedentity.TrackedEntityI
             {
                 trackedEntityInstance.getProgramOwners().add( new ProgramOwner( programOwner ) );
             }
-
         }
 
         if ( params.isDataSynchronizationQuery() )
         {
-            List<String> programs = trackedEntityInstance.getEnrollments().stream().map( e -> e.getProgram() ).collect( Collectors.toList() );
+            List<String> programs = trackedEntityInstance.getEnrollments().stream()
+                .map( Enrollment::getProgram )
+                .collect( Collectors.toList() );
+
+            readableAttributesCopy = new HashSet<>(
```

Comment:
Here you dont need to make a new HashSet, instead just use readableAttributeCopy = readableAttributeCopy.stream().filter()...

I think this is a bit cleaner, since readableAttributeCopy will actually be a copy all the way.